helix-core/src/syntax.rs

```diff
@@ -2024,6 +2024,57 @@ impl<I: Iterator<Item = HighlightEvent>> Iterator for Merge<I> {
     }
 }
 
+pub fn pretty_print_tree<W: fmt::Write>(fmt: &mut W, node: Node<'_>) -> fmt::Result {
+    pretty_print_tree_impl(fmt, node, true, None, 0)
+}
+
+fn pretty_print_tree_impl<W: fmt::Write>(
+    fmt: &mut W,
+    node: Node<'_>,
+    is_root: bool,
+    field_name: Option<&str>,
+    depth: usize,
+) -> fmt::Result {
+    fn is_visible(node: Node<'_>) -> bool {
```

Comment:
Specifying the anonymous lifetimes here is not necessary:

```suggestion
pub fn pretty_print_tree<W: fmt::Write>(fmt: &mut W, node: Node) -> fmt::Result {
    pretty_print_tree_impl(fmt, node, true, None, 0)
}

fn pretty_print_tree_impl<W: fmt::Write>(
    fmt: &mut W,
    node: Node,
    is_root: bool,
    field_name: Option<&str>,
    depth: usize,
) -> fmt::Result {
    fn is_visible(node: Node) -> bool {
```

Reply:
Added a fix! Thanks :)